src/State.cpp

```diff
@@ -123,7 +124,58 @@ prom_id_t tracer_state_t::enclosing_promise_id() {
          iterator != full_stack.rend(); ++iterator) {
         auto event = *iterator;
         if (event.type == stack_type::PROMISE)
-            return event.promise_id;    
+            return event.promise_id;
     }
     return -1;
 }
+
+std::string to_string(function_type f) {
+    switch (f) {
+        case function_type::CLOSURE:
+            return std::string("CLOSURE");
```

Comment:
style: `return "CLOSURE";` would be enough

Reply:
Will correct this.